### ISSUE TITLE
Bug in `ursina.audio` prevents loading audio files when in a different directory than root.

### ISSUE THREAD

**Issue by jgreenemi** [pokepetter/ursina#57]
## Bug Description

I understand there is some logic to try to help the user of `ursina.audio.Audio` to load audio without having to specify an extension. The logic that detects the desired file appears to have a bug when the file is not in a simple location (like placing the audio file in the project working directory along with the code that's running). In my case, I have audio resources separated into a different directory named `resources`. Using `ursina.audio.Audio` fails to load audio files from this directory, even though I've confirmed that the path is correct and reachable by other parts of my code. 

## Expected Behaviour

Take a look at the steps to reproduce to find the exact output. I would expect this to result in the audio file being played, but this is not the case. Further, I would expect such a failure to result in an exception being thrown, which was not the case here. This causes silent failures, which is unreliable in practice.

As a developer, being unable to simply pass in a filepath to a loader without it getting mangled behind the scenes is a significant hindrance, and was a poor introduction to an otherwise promising library.

### For others encountering this issue, a workaround.

To mitigate this issue, I had to recreate a bit of the `ursina.audio.Audio` class myself, pasting below to help anyone else who has been bit by this. If you are one such dev trying to resolve this issue, you'll find the solution simple, to just use the `ShowBase` loader directly:

```python
app = Ursina()

def load_audio(filepath: str) -> audio.Audio:
    a = audio.Audio(sound_file_name=None, loop=False)
    loaded_sound = app.loader.loadSfx(filepath)
    a.clip = loaded_sound
    return a

# Test:
load_audio(os.path.join('..', 'resources', 'select-long.wav')).play()
```

## Steps to Reproduce

Create a directory with the following structure:

```
+---project
    \---venv
    \---resources
    |   \---select-long.wav
    \---src
        \---__init__.py
        +---main.py
```

Follow the `ursina_for_dummies` tutorial up until loading sounds is mentioned, with your code being housed in the `src` package. At this point, add the following code:

```python
a = audio.Audio(sound_file_name=os.path.join('..', 'resources', 'select-long'), loop=False)
a.play()
```

Run this code with your current working directory as `project/src/`.

Observe the following error output, and observe no exception was thrown:

```
:audio(error): Cannot open file: urces\select-long.wav
:audio(error): Could not open audio urces\select-long.wav
```

The process should not crash in this state, but also should not play the audio file.

**Comment by pokepetter** [comment]
The recommended project structure is this:
```
    +---project
        \---resources
        |   \---select-long.wav
        +---main.py
```
    

By default ursina will load things from the root directory or a directory underneath. 
It doesn't normally make sense to hard code paths to external resources.
The reason for just using names to load assets is so you can sort assets into
folders or make compressed versions without having to change the code. 
    
If you really do need to change the root folder however, you can change application.asset_folder to the path you want.

I can't get it to error silently, but I'll look into making a more descriptive error message.

**Comment by jgreenemi** [comment]
Thank you for the quick response! I lifted the `main.py` out of `src` and passing in `os.path.join('resources', 'select-long')` now loads and plays the audio file as expected. For a number of reasons I will continue to use my original project structure, and keep the bug open as `os.path.join('..', 'resources', 'select-long')` is a valid path that shouldn't be failing, regardless if it's uncommon to reference audio assets in this way.

I wasn't aware of the existence of "application.asset_folder", that is something I'll look into to better understand the canonical approach to loading audio files, and possibly other assets. Where can I find documentation describing its use, where it gets consumed, etc.?

The context around the reasoning for using names is helpful. Can this be persisted somewhere in the code documentation? While the code explains the "how" of a decision it doesn't necessarily explain the "why". This benefit was not obvious to me as I didn't anticipate having to frequently change the format of existing resources, so much as changing their contents without changing the format (for example, it's more likely I'll change a `wav` to another `wav` with a different sound effect as opposed to changing a `wav` to `ogg`).

On the note about the error, are you saying you were unable to reproduce the error following the steps I provided? To be explicit, I want to know that a resource fails to load - but right now it appears to just print the fact that it failed to load, rather than actually throwing an exception. (The message could be more verbose about why it could not be loaded but that is beside the point.) I want to see an exception get thrown when a resource fails to load because this is an error state that needs to be handled. How a developer handles a resource loading failure in the code should be up to the developer, not swallowed by the library. Is this a behaviour from `ShowBase`, `Ursina`, or `Panda3d`?
